### ISSUE TITLE
started() and stopped()

### ISSUE THREAD

**Issue by AndreMaz** [moleculerjs/moleculer-template-project-typescript#7]
I think I found a solution to the `started()` and `stopped()` methods:

https://github.com/moleculerjs/moleculer-template-project-typescript/blob/ba8d5b8e190ca1057af86db8c40029d3505a34f9/template/services/greeter.service.ts#L72-L81

That currently shows the following problem:
`
  Type '() => void' is not assignable to type '() => Bluebird<void>'.
  Type 'void' is not assignable to type 'Bluebird<void>'.
`

The problem is in Moleculer's [`index.d.ts`](https://github.com/moleculerjs/moleculer/blob/23d04791d10faec4fd7a0f3086d38dc5bef12530/index.d.ts#L153-L154) that requires returning a Bluebird promise

```javascript
started?: () => Bluebird<void>;
stopped?: () => Bluebird<void>;
```
Because of this the only way to declare the `started` and `stopped` is 

```javascript
started() {
   return this.Promise.resolve() // this.Promise is a reference to Bluebird promise
}
```

However, it should be possible to use native promises and async/await. To make it work
`index.d.ts` must be updated as follows:
```javascript
started?: () => PromiseLike<void>;
stopped?: () => PromiseLike<void>;
```
With `PromiseLike` the compiler will accept anything that looks like a promise.

@faeron @icebob I think this change won't break anything but I'm not sure because I'm a complete newbie in Typescript

**Comment by faeron** [comment]
Nice find ! 👏👏   I didn't find that solution when porting the template, thats why they are commented out :D 
I also think that a change to the definition would not break anything.

**Comment by icebob** [comment]
I'm not a typescript guy, so I can't consider it, but please make a PR in moleculer repo and we will ask somebody to review it.

**Comment by AndreMaz** [comment]
Solved in [#422](https://github.com/moleculerjs/moleculer/pull/442)
